core/iwasm/common/wasm_runtime_common.c

```diff
@@ -7937,9 +7939,31 @@ wasm_runtime_disinherit_imports(WASMModuleCommon *module,
 #endif
 #if WASM_ENABLE_AOT != 0
     if (module->module_type == Wasm_Module_AoT) {
-        bh_assert(false && "Unsupported operation");
+        return aot_disinherit_imports((AOTModule *)module, imports,
+                                      import_count);
     }
 #endif
     LOG_ERROR("disinherit imports failed, invalid module type");
 }
 #endif /* WASM_ENABLE_LIB_WASI_THREADS != 0 || WASM_ENABLE_THREAD_MGR != 0 */
+
+const WASMExternInstance *
+wasm_runtime_get_extern_instance(const WASMExternInstance *imports,
+                                 uint32 import_count,
+                                 wasm_import_export_kind_t kind, uint32 index)
+{
+    for (uint32 i = 0, target_kind_index = 0; i < import_count; i++) {
+        if (imports[i].kind != kind) {
+            continue;
+        }
+
+        if (target_kind_index != index) {
+            target_kind_index++;
```

Comment:
a little confusing here, must the imports array be ordered? e.g. list of import globals, list of import functions, list of import memories, list of import tables?

Reply:
Yes. has to follow the order.

Reply:
Not sure whether it is consistent with wasm-c-api wasm_instance_new, must imports for wasm_instance_new be ordered too?
If yes, we had better add comment in wasm_export.h, e.g. struct InstantiationArgs?

Reply:
Check the update in wasm_export.h